File: scripts/r/calculo_indice_precio_ingreso_y_riesgo_de_burbuja.R
library(tidyverse)

df <- df %>%
  mutate(PIR = df$var1 / df$wage)

promedio_PIR <- mean(df$PIR)
desviacion_PIR <- sd(df$PIR)
umbral_riesgo <- promedio_PIR + (1.25 * desviacion_PIR)

df <- df %>%
  mutate(Riesgo = ifelse(PIR > umbral_riesgo, "Alto", "Bajo"))

plot(df$PIR, type = "l", col = "red")

write.csv(df, "df_riesgo.csv", row.names = FALSE)

File: scripts/python/calculo_indice_precio_ingreso_y_riesgo_de_burbuja.py
import pandas as pd
import matplotlib.pyplot as plt

df = df

df["PIR"] = df["var1"] / df["wage"]

promedio_PIR = df["PIR"].mean()
desviacion_PIR = df["PIR"].std()
umbral_riesgo = promedio_PIR + (1.25 * desviacion_PIR)

df["Riesgo"] = df["PIR"].apply(lambda x: "Alto" if x > umbral_riesgo else "Bajo")

plt.plot(df["PIR"], color="red")
plt.show()

df.to_csv("df_riesgo.csv", index=False)
